### PR TITLE
[search] TF-IDF for localities.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mpimenov @bykoianko PTAL

### CODE REVIEW

File: search/geocoder.cpp [hunk 1]
```diff
@@ -745,7 +706,7 @@ void Geocoder::FillLocalitiesTable(BaseContext const & ctx)
 
 #if defined(DEBUG)
         ft.GetName(StringUtf8Multilang::kDefaultCode, city.m_defaultName);
-        LOG(LDEBUG, ("City =", city.m_defaultName, radius));
+        LOG(LINFO, ("City =", city.m_defaultName, "radius =", radius, "tfidf =", l.m_tfidf));
```
Comment (bykoianko) [inline search/geocoder.cpp]: LDEBUG?

File: search/locality_scorer.cpp [hunk 2]
```diff
@@ -55,44 +109,34 @@ void LocalityScorer::GetTopLocalities(size_t limit, vector<Geocoder::Locality> &
...
+    return lhs.m_rank > rhs.m_rank;
+  });
+
+  std::vector<ExLocality> rs;
+  std::unordered_set<uint32_t> ids;
+
+  for (size_t i = 0; i < ls.size() && rs.size() < limit; ++i)
+  {
+    bool const inserted = ids.insert(ls[i].GetId()).second;
+    if (inserted)
```
Comment (bykoianko) [inline search/locality_scorer.cpp]: if (ids.insert(ls[i].GetId()).second) ?

File: search/geocoder.cpp (hunk 1 above)
Comment (ygorshenin) [inline search/geocoder.cpp]: Why? This code is compiled only when if defined(DEBUG), so there's no reason in LDEBUG.

File: search/locality_scorer.cpp (hunk 2 above)
Comment (ygorshenin) [inline search/locality_scorer.cpp]: This is the same, but more readable from my point of view.

File: search/geocoder.cpp (hunk 1 above)
Comment (bykoianko) [inline search/geocoder.cpp]: Yes, that's right. But are you sure that you want to have |LINFO| under #ifdef DEBUG?
At the first glance it'll be print in release version. I don't insist. It's just for your information.

File: search/locality_scorer.cpp (hunk 2 above)
Comment (bykoianko) [inline search/locality_scorer.cpp]: Ok.

File: search/geocoder.cpp (hunk 1 above)
Comment (ygorshenin) [inline search/geocoder.cpp]: Sorry? This code is not compiled in the release version.

File: search/search_tests/locality_scorer_test.cpp [hunk 3]
```diff
@@ -20,80 +25,77 @@ using namespace strings;
...
-  size_t const numTokens = params.GetNumTokens();
+    m_names[featureId].push_back(name);
+  }
 
-  for (size_t startToken = 0; startToken != numTokens; ++startToken)
+  void GetTopLocalities(size_t limit)
   {
-    for (size_t endToken = startToken + 1; endToken <= numTokens; ++endToken)
+    BaseContext ctx;
+    ctx.m_usedTokens.assign(m_params.GetNumTokens(), false);
```
Comment (bykoianko) [inline search/search_tests/locality_scorer_test.cpp]: Please add inline comment for false

Comment (ygorshenin) [inline search/search_tests/locality_scorer_test.cpp]: Could you please suggest the best comment?

Comment (bykoianko) [inline search/search_tests/locality_scorer_test.cpp]: Ok. Let's leave as is.

**Comment by ygorshenin** [comment]
The benefits from this request are quite arguable, we decided to temporarily freeze it.

**Pull request closed without merging**